Add instructions for installing Meshery on EKS cluster (#4090)
meshery/meshery: docs/pages/installation/eks.md

image: /assets/img/platforms/eks.png
---

{% include installation_prerequisites.html %}

## To set up and run Meshery on EKS:

> 

Reviewer: @leecalcote @navendu-pottekkat @alphaX86 would it be a nice UX to have separate instructions based on if the user wants to deploy Meshery in Docker or in k8s?
From my POV and with some hands-on it would be.

Reviewer: The only requirement when installing meshery "on" EKS using `mesheryctl system start` is that your kubeconfig is pointing to the EKS. An EKS user should know this because they would need this to use even simple kubectl command. However, it would be still nice to mention that this can be done using the command,
```
aws eks update-kubeconfig \
  --region <region-name> \
  --name <cluster-name>
```

> - Connect Meshery to your EKS cluster

Reviewer: ```suggestion
- [Install Meshery in Docker and connect to your EKS cluster](#link)
```

  - [Meshery CLI (mesheryctl)](#connect-meshery-to-elastic-kubernetes-service-cluster)
  - [eks CLI (eksctl)](https://eksctl.io/introduction/#installation)
  - [AWS CLI (aws)](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html)
- [Install Meshery on your EKS cluster](#install-meshery-into-your-eks-cluster)
- [Access Meshery's UI](#port-forward-to-the-meshery-ui)

### Connect Meshery to Elastic Kubernetes Service Cluster

> The following set of instructions expects you to have created a EKS cluster in your resource group Configure Meshery to connect to your EKS cluster by executing:

Reviewer: ```suggestion
**The following set of instructions are for deploying Meshery in Docker and connecting it to the EKS cluster in your resource group.**
You can configure Meshery to connect to your EKS cluster by executing:
```

 <pre class="codeblock-pre"><div class="codeblock">
 <div class="clipboardjs">
 mesheryctl system config eks
 </div></div>
 </pre>

#### Manual Steps

Alternatively, you may execute the following steps to manually configure Meshery to connect to your EKS cluster.

- Install [AWS CLI(aws)](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html), and login to your AWS account.
- After successfull login, you have to select the subscription with which your EKS is associated with by configuring your AWS CLI to your AWS account. Refer [this link](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for more

- Get/Update the kubeconfig from your EKS cluster by connecting with AWS CLI
<pre class="codeblock-pre"><div class="codeblock">
<div class="clipboardjs">
aws eks --region region update-kubeconfig --name cluster_name
</div></div>
</pre>

- Set your cluster context and check your cluster-info
<pre class="codeblock-pre"><div class="codeblock">
<div class="clipboardjs">
kubectl set-context EKS_SERVICE_NAME
kubectl cluster-info
</div></div>
</pre>

### Install Meshery into your EKS cluster

> 

Reviewer: Its important to mention these two here. Recently, many of the users are missing that they can install it using mesheryctl itself.
* Using mesheryctl
* Using helm

Author: Sadness... many cloud cluster instructions in our site aren't mentioned that we can do that

Author: Sounds a good idea 👍

image: /assets/img/platforms/eks.png
---

{% include installation_prerequisites.html %}

## To set up and run Meshery on EKS:

- Connect Meshery to your EKS cluster

>   - [Meshery CLI (mesheryctl)](#connect-meshery-to-azure-kubernetes-cluster)

Reviewer 2: Azure reference here.

  - [eks CLI (eksctl)](https://eksctl.io/introduction/#installation)
  - [AWS CLI (aws)](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html)
- [Install Meshery on your EKS cluster](#install-meshery-into-your-eks-cluster)
- [Access Meshery's UI](#port-forward-to-the-meshery-ui)

> ### Connect Meshery to Elastic Kubernetes Cluster

Reviewer 2: ```suggestion
### Connect Meshery to Elastic Kubernetes Service Cluster
```

The following set of instructions expects you to have created a EKS cluster in your resource group Configure Meshery to connect to your EKS cluster by executing:

 <pre class="codeblock-pre"><div class="codeblock">
 <div class="clipboardjs">
 mesheryctl system config eks
 </div></div>
 </pre>

> #### Manual Steps

Reviewer 2: Will you put the manual steps into a collapsed, but expandable section? Hopefully, users don't have to reference these.

Author: Collapsible? Hmm got it 👍

> Alternatively, you may execute the following steps to manually configure Meshery to connect to your AKS cluster.

Reviewer 2: ```suggestion
Alternatively, you may execute the following steps to manually configure Meshery to connect to your EKS cluster.
```

- Install [AWS CLI(aws)](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html), and login

>   to your AWS account. Install [EKS CLI (eksctl)](https://eksctl.io/introduction/#installation) too to access eks cluster easily.

Reviewer 2: I might be missing it - where is `eksctl` used in these instructions? If it isn't, please remove this reference.

Author: Eksctl isn't used but just came across... Anyway I'll remove it

- After successfull login, you have to select the subscription with which your AKS is associated with by configuring your AWS CLI to your AWS account. Refer [this link](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for more

- Get/Update the kubeconfig from your EKS cluster by connecting with AWS CLI
<pre class="codeblock-pre"><div class="codeblock">
<div class="clipboardjs">
aws eks --region region update-kubeconfig --name cluster_name
</div></div>
</pre>

- Set your cluster context and check your cluster-info
<pre class="codeblock-pre"><div class="codeblock">
<div class="clipboardjs">
kubectl set-context EKS_SERVICE_NAME

> kubectl cluster-info

Reviewer 2: Does Meshery use this info? If not, please only list instructions necessary for installing Meshery. Users can refer to EKS docs for how to connect to EKS in general.

</div></div>
</pre>

> ### Install Meshery into your AKS cluster

Reviewer 2: ```suggestion
### Install Meshery into your EKS cluster
```

 <pre class="codeblock-pre"><div class="codeblock">
 <div class="clipboardjs">
 $ kubectl create ns meshery
 $ helm repo add meshery https://meshery.io/charts/
 $ helm install meshery --namespace meshery
 </div></div>
 </pre>

>  - Meshery server supports customizing authentication flow callback URL, which can be configured in the following way

Reviewer 2: Good. Customizing callback should be its own section and should not hidden in the collapsed manual steps.

 <pre class="codeblock-pre"><div class="codeblock">
 <div class="clipboardjs">
 $ helm install meshery --namespace meshery --set env.MESHERY_SERVER_CALLBACK_URL=https://custom-host meshery/meshery
 </div></div>
 </pre>

### Port forward to the Meshery UI

 <pre class="codeblock-pre"><div class="codeblock">
 <div class="clipboardjs">
 export POD_NAME=$(kubectl get pods --namespace meshery -l "app.kubernetes.io/name=meshery,app.kubernetes.io/instance=meshery" -o jsonpath="{.items[0].metadata.name}")

>  kubectl --namespace meshery port-forward $POD_NAME 9081:8080

Reviewer 2: ```suggestion
 kubectl --namespace meshery port-forward $POD_NAME 9081:9081
```

Please change the 8080 reference elsewhere in the docs to 9081, too.

 </div></div>
 </pre>

> Meshery should now be running in your EKS cluster and the Meshery UI should be locally accessible. Navigate to [http://localhost:9081](http://localhost:9081) to log into Meshery.

Reviewer 2: No, Meshery UI will be remotely accessible at whatever public (or private) service endpoint they have exposed it on.

Author: Oh :) seems I need to change that in Azure reference as well. Or no need?

Reviewer 2: Yes, EKS and AKS run very similarly.
